docs/jaxpr.rst

```diff
@@ -0,0 +1,491 @@
+Understanding JAXPR
```

Comment:
I'm a little concerned that these docs will either become stale, or else will slow down our ability to revise jaxprs (as we need to keep these docs updated).

How about including a date and a git hash at the top of this doc, along with an explanation that the description here applies to jaxprs for that git commit of JAX? That way we can keep this as updated as possible while still being transparent about any staleness.

Reply:
I have added a date and commit number. 

These documents may become stale, although we should try hard to update them. 
Updating them should not really slow us down in any noticeable way, and will allow others perhaps to help (including updating the docs).